src/ape/managers/compilers.py

```diff
@@ -51,11 +51,8 @@ def registered_compilers(self) -> Dict[str, CompilerAPI]:
             compiler = compiler_class(config=config)
 
             for extension in extensions:
-
-                if extension in registered_compilers:
```

Comment:
I am not sure this could ever happen. And if it did, I think this would still confuse the user